src/boss/azure_worker.go

```diff
@@ -0,0 +1,214 @@
+package boss
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/user"
+	"sync"
+	"time"
+)
+
+type AzureWorkerPool struct {
+	workerNum int
+	workers   *map[string]*AzureWorker
+	nextId    int
+}
+
+type AzureWorker struct {
+	pool         *AzureWorkerPool
+	workerId     string
+	configPosit  int
+	diskName     string
+	vnetName     string
+	subnetName   string
+	nsgName      string
+	nicName      string
+	publicIPName string
+	privateAddr  string
+	publicAddr   string
+}
+
+func NewAzureWorkerPool() *WorkerPool {
+	conf, err := ReadAzureConfig()
+	if err != nil {
+		log.Fatalln(err)
+	}
+	num := conf.Resource_groups.Rgroup[0].Numvm
+	workers := make(map[string]*AzureWorker, num)
+	pool := &AzureWorkerPool{
+		workerNum: num,
+		workers:   &workers,
+		nextId:    num + 1,
+	}
+	for i := 0; i < num; i++ {
+		worker_i := new(AzureWorker)
```

Comment:
Instead of initializing one by one, let's use this:

worker_i := &AzureWorker{
    pool = ...
}